catalog-rest-service/src/main/resources/ui/src/components/Modals/ModalWithMarkdownEditor/ModalWithMarkdownEditor.tsx

```diff
@@ -30,9 +30,9 @@ type Props = {
   header: string;
   value: string;
   placeholder: string;
-  onSave: (text: string) => void;
+  onSave?: (text: string) => void;
```

Comment:
Why do we need to make this optional?

Reply:
@darth-coder00 because we don't want to pass props unnecessarily if the component is readOnly.